qcodes_contrib_drivers/drivers/RohdeSchwarz/SMW200A.py

```diff
@@ -0,0 +1,2313 @@
+
+        self.add_parameter('mode',
+                           label='Mode',
+                           set_cmd='SOUR{}:'.format(self.hwchan)+'IQ:OUTP:ANAL:MODE {}',
+                           get_cmd='SOUR{}:'.format(self.hwchan)+'IQ:OUTP:ANAL:MODE?',
+                           vals=mode_validator,
+                           docstring="""
+                           Determines the mode for setting the output parameters. Values are:
+                               'FIX': Locks the I/Q output settings
+                               'VAR': Unlocks the settings (only available with option SMW-K16)
+                           """)
+
+        if 'SMW-B10' in self._parent.options:
+            level_validator = vals.Numbers(0.04, 4)
+        else: #option SMW-B9
+            level_validator = vals.Numbers(0.04, 2)
+        self.add_parameter('level',
+                           label='Level',
+                           set_cmd='SOUR{}:'.format(self.hwchan)+'IQ:OUTP:LEV {}' \
+                                   if self.mode() == 'VAR' else False,
+                           get_cmd='SOUR{}:'.format(self.hwchan)+'IQ:OUTP:LEV?',
+                           get_parser=float,
+                           vals=level_validator,
+                           unit='V',
+                           docstring="Sets the off-load voltage Vp of the analog I/Q signal"
+                                     " output. Values are in range 0.04V to 4V for option"
+                                     " SMW-B10 and in range 0.04V to 2V for option SMW-B9."
+                                     " The value range is adjusted so that the maximum"
+                                     " overall output voltage does not exceed 4V. Only"
+                                     " settable when mode has the value 'VAR'.")
+
+        self.add_parameter('coupling',
+                           label='Coupling',
+                           set_cmd='SOUR{}:'.format(self.hwchan)+'IQ:OUTP:ANAL:BIAS:COUP:STAT {}',
+                           get_cmd='SOUR{}:'.format(self.hwchan)+'IQ:OUTP:ANAL:BIAS:COUP:STAT?',
+                           val_mapping={'ON': 1, 'OFF': 0},
+                           vals=vals.Enum('ON', 'OFF'),
+                           docstring="Couples the bias setting of the I and Q signal"
+                                     " components. Values are 'ON and 'OFF'.")
+
+        if 'SMW-B10' in self._parent.options:
+            lower = -4+self.level()/2+self.i_offset()/2
+            upper = 4-self.level()/2-self.i_offset()/2
+            bias_validator = vals.Numbers(lower, upper)
+        else: #option SMW-B9
+            bias_validator = vals.Numbers(-0.2, 2.5)
+        self.add_parameter('i_bias',
+                           label='I bias',
+                           set_cmd='SOUR{}:'.format(self.hwchan)+'IQ:OUTP:ANAL:BIAS:I {}' \
+                                   if self.mode() == 'VAR' else False,
```

Comment:
Is there a better way to write this? I am not comfortable with having if-else condition in a set_cmd. May be we can have a private method for these type of parameters?